stdlib/public/core/StringBridge.swift

```diff
@@ -154,30 +136,9 @@ extension String {
 
     let length = _swift_stdlib_CFStringGetLength(cfImmutableValue)
 
-    // Look first for null-terminated ASCII
-    // Note: the code in clownfish appears to guarantee
-    // nul-termination, but I'm waiting for an answer from Chris Kane
-    // about whether we can count on it for all time or not.
-    let nulTerminatedASCII = _swift_stdlib_CFStringGetCStringPtr(
-      cfImmutableValue, kCFStringEncodingASCII)
-
-    // start will hold the base pointer of contiguous storage, if it
-    // is found.
-    var start: UnsafeMutableRawPointer?
-    let isUTF16 = (nulTerminatedASCII == nil)
-    if isUTF16 {
-      let utf16Buf = _swift_stdlib_CFStringGetCharactersPtr(cfImmutableValue)
-      start = UnsafeMutableRawPointer(mutating: utf16Buf)
-    } else {
-      start = UnsafeMutableRawPointer(mutating: nulTerminatedASCII)
-    }
-
     self._core = _StringCore(
```

Comment:
'fraid not.  If the string is ASCII, we try to store it as such, and avoid branching by using math to adjust our accesses to use 1 or 2 bytes at a time.